src/librustc/mir/visit.rs

```diff
@@ -198,7 +198,7 @@ macro_rules! make_mir_visitor {
             }
 
             fn visit_ty(&mut self,
-                        ty: & $($mutability)? Ty<'tcx>,
+                        ty: $(& $mutability)? Ty<'tcx>,
```

Comment:
Wow, that's a neat trick! cc @nikomatsakis how did we not think of this?

Reply:
`{visit,super}_{local,region,const}` (and maybe a few others?) should probably get the same treatment.

Reply:
heh, a good question.